client/src/utils/collisions.ts

```diff
@@ -0,0 +1,120 @@
+import { ContextualizedLesson } from '../components/Lessons';
+import { intervalFromLesson } from './TimeUtil';
+
+type CompareKeyFunction<T> = (x: T) => number;
+type CompareFunction<T> = (a: T, b: T) => number;
+const chainComparisons =
+  <T>(funcs: CompareFunction<T>[]): CompareFunction<T> =>
```

Comment:
**note (non-blocking):** TS has a surprisingly nice way of dealing with variadic arguments, so you could do this and save on one array by using this as the param instead
```
...funcs: CompareFunction<T>[]
```